bin/plotting/pycbc_plot_bank_corner

```diff
@@ -0,0 +1,286 @@
+    if cpar and co in cpar:
+        required_params.append(co)
+
+# Check for possible double-counting of duration parameter:
+if 'duration' in required_params and any([p.endswith('_duration')
+                                          for p in required_params]):
+    # Check whether duration was specified directly:
+    dur_pars = [par for par in args.parameters if 'duration' in par]
+    _dur_pars = [par for par in args.parameters if '_duration' in par]
+    if len(dur_pars) == len(_dur_pars):
+        # It looks like duration has been added where it didnt need to be
+        required_params.remove('duration')
+
+logging.info("Required parameters to get from bank: %s",
+             ', '.join(required_params))
+
+# Do the same with the duration functions, but here we need to make
+# sure we have some other keys in order to calculate duration
+duration_required_keys = ['mass1', 'mass2', 'spin1z',
+                          'spin2z', 'f_lower']
+
+if any(['duration' in par for par in args.parameters]):
+    required_params += duration_required_keys
+
+# Some things may have been double counted - undo this
+required_params = np.unique(required_params)
+
+# Get parameters not directly in the bank:
+for p in required_params:
+    if p in bank: continue
+    if p not in bconv.conversion_options:
+        raise KeyError(f"ERROR Parameter {p} not in bank or conversion "
+                       "options, choose from bank parameters or " + \
+                       ', '.join(bconv.conversion_options))
+
+    logging.info("Converting %s", p)
+    bank[p] = bconv.get_bank_property(p, bank, np.arange(banklen))
+
+# Durations can be negative where get_imr_duration is not perfect - 
+# set a minimum value of 0.15s
+if 'duration' in bank:
+    bank['duration'] = np.maximum(bank['duration'], 0.15)
+if 'template_duration' in bank:
+    bank['template_duration'] = np.maximum(bank['template_duration'], 0.15)
+
+if args.mpl_style == 'dark_background':
+    hist_color = 'white'
+else:
+    hist_color = 'black'
+
+# All parameters should be in the bank now, check they are the right size:
+assert all([len(bank[p]) == banklen for p in bank])
+
+# Change the bank to the format required for the plot:
+bank_fa = FieldArray.from_arrays(bank.values(), names=list(bank.keys()))
+
+# Check that the required parameters have min/max values set:
+required_minmax = args.parameters[:]
+if cpar:
+    required_minmax.append(cpar)
+
+# check for min/max values of the color parameter
+for p in required_minmax:
+    if p not in mins:
+        mins[p] = bank_fa[p].min()
+    if p not in maxs:
+        maxs[p] = bank_fa[p].max()
+
+# Deal with non-coloring case:
+zvals = bank_fa[cpar] if cpar else None
+
+logging.info("Generating corner plot")
+
+fig, axis_dict = create_multidim_plot(
+    args.parameters,
+    bank_fa,
+    labels=args.parameters_labels,
+    plot_marginal=args.plot_histogram,
+    plot_scatter=True,
+    plot_contours=False,
+    scatter_cmap="rainbow",
```

Comment:
I would avoid the rainbow colormap because of its [weird lightness behavior](https://matplotlib.org/stable/_images/sphx_glr_colormaps_014.png). Use one of the [perceptually uniform](https://matplotlib.org/stable/_images/sphx_glr_colormaps_008.png) colormaps instead.

Reply:
Updated to use plasma - many others have points with brightness that goes too close to white (in my opinion)